homeassistant/components/tuneblade_remote/media_player.py

```diff
@@ -0,0 +1,161 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up TuneBlade Remote from a config entry."""
+    coordinator: TuneBladeDataUpdateCoordinator = config_entry.runtime_data
+
+    runtime_data = hass.data.setdefault(DOMAIN, {}).setdefault(
+        config_entry.entry_id, {}
+    )
+    added_ids = runtime_data.setdefault("added_ids", set())
+
+    entities_master = []
+    entities_others = []
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id == MASTER_ID:
+            entities_master.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id != MASTER_ID:
+            entities_others.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    if entities_master:
+        async_add_entities(entities_master, True)
+
+    if entities_others:
+        async_add_entities(entities_others, True)
+
+
+class TuneBladeMediaPlayer(TuneBladeEntity, MediaPlayerEntity):
+    """Media player for TuneBlade devices."""
+
+    def __init__(
+        self,
+        coordinator: TuneBladeDataUpdateCoordinator,
+        device_id: str,
+        device_data: dict[str, Any],
+    ) -> None:
+        """Initialize the media player entity."""
+        super().__init__(
+            coordinator,
+            coordinator.config_entry,
+            device_id,
+            device_data.get("name"),
+        )
+        self.device_id = device_id
+        self._attr_name = device_data["name"]
+        safe_name = self._attr_name.replace(" ", "_")
+        self._attr_unique_id = f"{device_id}@{safe_name}"
+        self._attr_volume_level = None
+        self._attr_state = MediaPlayerState.OFF
+        self._attr_supported_features = (
+            MediaPlayerEntityFeature.TURN_ON
+            | MediaPlayerEntityFeature.TURN_OFF
+            | MediaPlayerEntityFeature.VOLUME_SET
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        return self.device_id in self.coordinator.data
+
+    async def async_turn_on(self) -> None:
+        """Turn on the media player."""
+        await self.coordinator.client.connect(self.device_id)
+        await asyncio.sleep(5)
+        await self.coordinator.async_request_refresh()
+        self.hass.async_create_task(self._delayed_refresh())
+
+    async def async_turn_off(self) -> None:
+        """Turn off the media player."""
+        await self.coordinator.client.disconnect(self.device_id)
+        await asyncio.sleep(5)
+        await self.coordinator.async_request_refresh()
+        self.hass.async_create_task(self._delayed_refresh())
```

Comment:
Why do we sleep and why do we do the delayed refresh? We now do a request, sleep 5 seconds, update state, sleep 5 seconds, update state. Why?